src/main/java/duke/iomethods/IOMethods.java

```diff
@@ -0,0 +1,97 @@
+package duke.iomethods;
+
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+
+public class IOMethods {
+    public static ArrayList<String> splitToTwo(String line, String delimiter) {
+        ArrayList<String> words = new ArrayList<>();
+        int divider = line.indexOf(delimiter);
+
+        words.add(line);
+
+        if (divider != -1) {
+            words.set(0, line.substring(0, divider));
+            words.add(line.substring(divider+1));
+        }
+        return words;
+    }
+
+    public static void printWithDivider(String stringWithinDivider) {
+        String breakLine = "\t____________________________________________________________";
+        System.out.println(breakLine);
+        stringWithinDivider = stringWithinDivider.replace("\n", "\n\t");
+        System.out.println("\t" + stringWithinDivider);
+        System.out.println(breakLine);
+    }
+
+    public static void errorHandler(String input) throws DukeException {
+        String[] words = input.split(" ");
+        String command = words[0];
+
+        switch (command) {
+        case "todo":
+        case "event":
+        case "deadline":
+            String taskName = getNextWord(input, command);
+            if (taskName.equals("")) {
+                String errorMsg = String.format("%s requires a name\n", command);
+                throw new DukeException(errorMsg);
+            } else {
+                if (command.equals("event") || command.equals("deadline")) {
+                    int indexOfSlash = input.indexOf("/");
+                    String date = indexOfSlash == -1 ? "" : input.substring(indexOfSlash);
+                    if (date.length() <= 1) {
+                        String errorMsg = String.format("%s requires a valid date in the format taskName /date\n", command);
+                        throw new DukeException(errorMsg);
+                    }
+                }
+            }
```

Comment:
It is strange that `event` and `deadline` blocks of code are handled here after the fallthrough and there seems to be no course of action for `todo` beside checking for empty taskName. These lines seems [trippy](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-trip-up-reader) and you could try to restructure it to a [simple](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#practice-kissing) form.